salt/odl-video/s3_buckets_cloudfront.sls

```diff
@@ -0,0 +1,112 @@
+{% set odl_video_bucket_prefix = 'odl-video-service' %}
+{% set odl_video_bucket_suffix = ['ci', 'rc', 'prod'] %}
+{% set odl_video_bucket_purposes = ['thumbnails', 'transcoded', 'uswitch', 'watch', 'dist'] %}
+
+{% for bucket_sffix in odl_video_bucket_suffix %}
+{% for bucket_purpose in odl_video_bucket_purposes %}
+create_{{ bucket_prefix}}-{{ bucket_purpose }}-{{ bucket_suffix }}:
+  boto_s3_bucket.present:
+    - Bucket: {{ bucket_prefix}}-{{ bucket_purpose }}-{{ bucket_suffix }}
+    - region: us-east-1
+    - CORSRules:
+      - AllowedOrigin: ["*"]
+        AllowedMethod: ["GET"]
+        AllowedHeader: ["Authorization"]
+        MaxAgeSconds: 3000
+    - Versioning:
+        Status: "Enabled"
+{% endfor %}
+{% endfor %}
+
+{% for odl_video_bucket_suffix in ['rc', 'prod'] %}
+create_cloudfront_distribution_{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_suffix }}:
+  boto_cloudfront.present:
+  - name: {{ odl_video_bucket_prefix }}-{{ odl_video_bucket_suffix }}
+  - config:
+      CacheBehaviors:
+        Items:
+        - AllowedMethods:
+            CachedMethods:
+              Items:
+              - HEAD
+              - GET
+              - OPTIONS
+            Items:
+            - HEAD
+            - GET
+            - OPTIONS
+          Compress: false
+          DefaultTTL: 86400
+          ForwardedValues:
+            Cookies:
+              Forward: none
+            Headers:
+              Items:
+              - Access-Control-Request-Headers
+              - Access-Control-Request-Method
+              - Origin
+            QueryString: false
+          MaxTTL: 31536000
+          MinTTL: 0
+          PathPattern: /{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}*
+          SmoothStreaming: false
+          TargetOriginId: S3-{{ odl_video_bucket_prefix }}-{{ odl_video_bucket_purposes }}-{{ odl_video_bucket_suffix }}
```

Comment:
It looks like you either meant to make this part of a for loop or only target one of the buckets for the origin. As this is right now it will attempt to set the origin ID to something like `S3-odl-video-services-['thumbnails', 'transcoded',...etc.]-rc`

Reply:
Changed by adding an additional for loop